lib/render/pxrUsdMayaGL/batchRenderer.cpp

```diff
@@ -1196,6 +1198,8 @@ UsdMayaGLBatchRenderer::_ComputeSelection(
 
     _selectResults.clear();
 
+    UBOBindingsSaver bindingsSaver;
```

Comment:
This version is slightly different from the previous version in that the instantiation of the `UBOBindingsSaver` happens **outside** of `_TestIntersection()`. In relatively infrequent cases, the loop below may go through more than one iteration, so this avoids repeatedly saving and restoring the bindings on each iteration.

This is also why a `UBOBindingsSaver` was needed in `TestIntersectionCustomPrimFilter()` above as well.

The original version of this PR is available for comparison in commit 37d2924fb875037539ac38a13034d052606d0ff1.